numpy/core/src/multiarray/ctors.c

```diff
@@ -2335,6 +2339,23 @@ PyArray_FromInterface(PyObject *origin)
         else {
             base = origin;
         }
+#if defined(NPY_PY3K)
+        if (PyObject_GetBuffer(base, &view, PyBUF_WRITABLE) < 0) {
+            PyErr_Clear();
+            if (PyObject_GetBuffer(base, &view, PyBUF_SIMPLE) < 0) {
+                goto fail;
+            }
+            dataflags &= ~NPY_ARRAY_WRITEABLE;
+        }
+        data = (char *)view.buf;
+        /*
+         * XXX: PyObject_AsWriteBuffer does also this, but it is unsafe, as
```

Comment:
Mostly looking through the PR out of curiosity - and this comment confused me as `PyObject_AsWriteBuffer` is deprecated in python3; maybe clarify that the "The release below is unsafe, but the python2 PyObject_AsWriteBuffer does it as well.... etc."